Implement UTXOSetValue to be used by UTXOSet

MichaelKim20 opened the pull request:
Add freezing capability to UTXO #204 of part

To freeze UTXO, add the following information to UTXOSet:
The height of block.
The type of transaction.
The previous type of transaction.

MichaelKim20 commented:
### Transaction & UTXO in Freeze
The design was based on **Definition of Done**.
Illustrated the changing situation over time for **Block / Transaction / UTXOSet.**

<img width="928" alt="Transaction & UTXO in Freeze" src="https://user-images.githubusercontent.com/11639939/64497915-a2fb3e00-d2ec-11e9-808e-231aafdad1b9.png">

AndrejMitrovic commented on source/agora/consensus/data/UTXOSet.d:
```diff
@@ -80,23 +151,92 @@ public class UTXOSet
+                auto utxo_hash = getHash(input.previous, input.index);
+                UTXOSetValue prev_utxo;
+
+                if (this.findUTXOSetValue (input.previous, input.index, prev_utxo))
```
~This seems to look for a previous frozen UTXO when a new frozen UTXO is added to the cache. So how do you add the very first frozen UTXO? It won't find a previous one, because a previous one doesn't exist yet.~

The pull request passes the test-suite, but there are no tests for any of this code.

AndrejMitrovic replied:
Ah I was wrong, I see it in the code below. But, could you add tests?

MichaelKim20 commented:
Ready for review

MichaelKim20 committed:
Implement UTXOSetValue for freezing

UTXOSetValue is used by UTXOSet
To freeze UTXO, add the following information to UTXOSet:
The height of block.
The type of transaction.

Co-Authored-By: HyeonyeobKim <hyeonyeob.kim@bpfkorea.org>

MichaelKim20 commented:
Proceed to PR #326.
Current PR is closed.
